File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string,int> mp;
    for (int i=0; i<clothes.size(); i++){
        mp[clothes[i][1]]++;
    }
    for (auto a : mp){
        answer *= (a.second + 1);
    }
    answer--;
    return answer;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String,Integer> mp = new HashMap<>();
        for (int i=0; i< clothes.length; i++){
            if (mp.get(clothes[i][1]) == null || mp.get(clothes[i][1]) == 0){
                mp.put(clothes[i][1], 1);
            }
            else mp.put(clothes[i][1], mp.get(clothes[i][1]) + 1);
        }
        for(String key : mp.keySet()){
          answer *= mp.get(key) + 1;
        }
        answer--;
        return answer;
    }
}